channels/set.go

```diff
@@ -27,6 +27,15 @@ const UserStarChannel = "*"     // user channel for "can access all docs"
 const DocumentStarChannel = "!" // doc channel for "visible to all users"
 const AllChannelWildcard = "*"  // wildcard for 'all channels'
 
+type ID struct {
+	Name         string // name of channel
+	CollectionID uint32 // collection it belongs to
+}
+
+type Set map[ID]present
```

Comment:
How efficient is go at doing map looking based on struct keys?  I don't think I've tested, might be interesting to see a simple benchmark of string vs ID lookups for different sized maps.